src/sys/unix.rs

```diff
@@ -519,15 +526,14 @@ fn recvmsg(
     } else {
         size_of::<libc::sockaddr_storage>() as libc::socklen_t
     };
-    let mut msg = libc::msghdr {
-        msg_name: msg_name.cast(),
-        msg_namelen,
-        msg_iov: bufs.as_mut_ptr().cast(),
-        msg_iovlen: min(bufs.len(), IovLen::MAX as usize) as IovLen,
-        msg_control: ptr::null_mut(),
-        msg_controllen: 0,
-        msg_flags: 0,
-    };
+    let mut msg: libc::msghdr = unsafe { mem::zeroed() };
```

Comment:
Can you revert back to creating the structure directly, it makes a bit more awkward with differences between OSs I know, but it forces you to check the structure when porting to a new OS (like what happend here).

Reply:
The trouble is that this structure has private padding fields on Fuchsia, and can't be instantiated directly.

Shall I add a comment? Is there a better solution?

Reply:
Does something like the following work?
```
msghdr {
    // fields ...
    .. Default::default(),
}
```

Or does `msghdr` need to implement `Default` for that? I forgot...

If it's not possible, keep the change as is but with a comment indeed. And can you remove the `msg_controllen` and `msg_flags` fields as there already zero.

Reply:
Yeah, it has to impl `Default`. Added a comment and removed zero initialization.